tests/ci/integration/mariadb_patch/0002-WIP-Stop-using-the-unchecked-version-of-DES_set_key.patch

```diff
@@ -0,0 +1,115 @@
+                                                                          ^
+make[2]: *** [sql/CMakeFiles/sql.dir/des_key_file.cc.o] Error 1
+
+Example error on Fedora:
+
+[ 83%] Building CXX object sql/CMakeFiles/sql.dir/des_key_file.cc.o
+/builds/CPACK_BUILD_SOURCE_DIRS_LONG_NAME_REQUIREMENT/sql/des_key_file.cc: In function 'bool load_des_key_file(const char*)':
+/builds/CPACK_BUILD_SOURCE_DIRS_LONG_NAME_REQUIREMENT/sql/des_key_file.cc:84:10: error: cannot convert 'DES_cblock' {aka 'DES_cblock_st'} to 'uint8_t*' {aka 'unsigned char*'}
+   84 |          ivec);
+      |          ^~~~
+      |          |
+      |          DES_cblock {aka DES_cblock_st}
+In file included from /usr/local/include/openssl/pem.h:62,
+                 from /usr/local/include/openssl/ssl.h:149,
+                 from /builds/CPACK_BUILD_SOURCE_DIRS_LONG_NAME_REQUIREMENT/include/violite.h:152,
+                 from /builds/CPACK_BUILD_SOURCE_DIRS_LONG_NAME_REQUIREMENT/sql/des_key_file.h:22,
+                 from /builds/CPACK_BUILD_SOURCE_DIRS_LONG_NAME_REQUIREMENT/sql/des_key_file.cc:18:
+/usr/local/include/openssl/cipher.h:350:44: note:   initializing argument 8 of 'int EVP_BytesToKey(const EVP_CIPHER*, const EVP_MD*, const uint8_t*, const uint8_t*, size_t, unsigned int, uint8_t*, uint8_t*)'
+  350 |                                   uint8_t *iv);
+      |                                   ~~~~~~~~~^~
+/builds/CPACK_BUILD_SOURCE_DIRS_LONG_NAME_REQUIREMENT/sql/des_key_file.cc:85:2: error: 'DES_set_key_unchecked' was not declared in this scope
+   85 |  DES_set_key_unchecked(&keyblock.key1,&(des_keyschedule[(int)offset].ks1));
+      |  ^~~~~~~~~~~~~~~~~~~~~
+make[2]: *** [sql/CMakeFiles/sql.dir/build.make:177: sql/CMakeFiles/sql.dir/des_key_file.cc.o] Error 1
+---
+ sql/des_key_file.cc | 10 +++++-----
+ sql/item_strfunc.cc | 16 ++++++++--------
+ 2 files changed, 13 insertions(+), 13 deletions(-)
+
+diff --git a/sql/des_key_file.cc b/sql/des_key_file.cc
+index bfbe04f6015..084523a6d4f 100644
+--- a/sql/des_key_file.cc
++++ b/sql/des_key_file.cc
+@@ -78,13 +78,13 @@ load_des_key_file(const char *file_name)
+ 	DES_cblock ivec;
+ 	bzero((char*) &ivec,sizeof(ivec));
+ 	// We make good 24-byte (168 bit) key from given plaintext key with MD5
+-	EVP_BytesToKey(EVP_des_ede3_cbc(),EVP_md5(),NULL,
++	EVP_BytesToKey( EVP_des_ede3_cbc(),EVP_md5(),NULL,
+ 		       (uchar *) start, (int) (end-start),1,
+ 		       (uchar *) &keyblock,
+-		       ivec);
+-	DES_set_key_unchecked(&keyblock.key1,&(des_keyschedule[(int)offset].ks1));
+-	DES_set_key_unchecked(&keyblock.key2,&(des_keyschedule[(int)offset].ks2));
+-	DES_set_key_unchecked(&keyblock.key3,&(des_keyschedule[(int)offset].ks3));
++                       ivec.bytes);
++	DES_set_key(&keyblock.key1,&(des_keyschedule[(int)offset].ks1));
++	DES_set_key(&keyblock.key2,&(des_keyschedule[(int)offset].ks2));
++	DES_set_key(&keyblock.key3,&(des_keyschedule[(int)offset].ks3));
```

Comment:
Did you consider adding a simple function that maps DES_set_key_unchecked -> DES_set_key? This isn't the only project to hit this issue.

Reply:
I took this patch directly from the work ottoke@ had been doing. But I can add it in a subsequent PR and remove this patch once we do so